src/components/documentManager.ts

```diff
@@ -0,0 +1,678 @@
+        // no showItemId for cc audio
+        if (getActiveDeviceId() === deviceIds.AUDIO) return;
+
+        const item: BaseItemDto = await JellyfinApi.authAjaxUser(
+            'Items/' + itemId,
+            {
+                dataType: 'json',
+                type: 'GET'
+            }
+        );
+
+        DocumentManager.showItem(item);
+    }
+
+    /**
+     * Get HTML content used to display the rating of an item
+     *
+     * @param {BaseItemDto} item to look up
+     * @returns {string} html to put in document
+     */
+    private static getRatingHtml(item: BaseItemDto): string {
+        let html = '';
+        if (item.CommunityRating != null) {
+            html +=
+                `<div class="starRating" title="${item.CommunityRating}"></div>` +
+                '<div class="starRatingValue">' +
+                item.CommunityRating.toFixed(1) +
+                '</div>';
+        }
+
+        if (item.CriticRating != null) {
+            const verdict = item.CriticRating >= 60 ? 'fresh' : 'rotten';
+            html +=
+                `<div class="${verdict} rottentomatoesicon" title="${verdict}"></div>` +
+                `<div class="criticRating">${item.CriticRating}%</div>`;
+        }
+
+        return html;
+    }
+
+    /**
+     * Set the status of the app, and switch the visible view
+     * to the corresponding one.
+     *
+     * @param {string} status to set
+     */
+    public static setAppStatus(status: string): void {
+        this.status = status;
+        document.body.className = status;
+    }
+
+    /**
+     * Get the status of the app
+     *
+     * @returns {string} app status
+     */
+    public static getAppStatus(): string {
+        return this.status;
+    }
+
+    /**
+     * BACKDROP LOGIC
+     *
+     * Backdrops are set on the waiting container.
+     * They are switched around every 30 seconds by default
+     * (governed by startBackdropInterval)
+     *
+     * @param {BaseItemDto | null} item Item to use for waiting backdrop, null to remove it.
+     */
+    public static async setWaitingBackdrop(
+        item: BaseItemDto | null
+    ): Promise<void> {
+        // no backdrop as a fallback
+        let src: string | null = null;
+
+        if (item != null) {
+            if (item.BackdropImageTags && item.BackdropImageTags.length) {
+                // get first backdrop of image if applicable
+                src = JellyfinApi.createUrl(
+                    `Items/${item.Id}/Images/Backdrop/0?tag=${item.BackdropImageTags[0]}`
+                );
+            } else if (
+                item.ParentBackdropItemId &&
+                item.ParentBackdropImageTags &&
+                item.ParentBackdropImageTags.length
+            ) {
+                // otherwise get first backdrop from parent
+                src = JellyfinApi.createUrl(
+                    `Items/${item.ParentBackdropItemId}/Images/Backdrop/0?tag=${item.ParentBackdropImageTags[0]}`
+                );
```

Comment:
Can't we also have a function that takes an ID, Image Type and tag and returns us this kind of url without needing it to be scattered all around?

Reply:
It seems like this one just takes backdrop number 0, ideally we would pick a random one, I'd like to express that too, somehow